File: app/src/main/java/manabboro/alarmclock/database/AlarmDao.java
package manabboro.alarmclock.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Room;
import androidx.room.Update;

import java.util.List;

import manabboro.alarmclock.model.Alarm;

@Dao
public interface AlarmDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Alarm alarm);

    @Query("DELETE FROM alarm_table")
    void deleteAll();

    @Query("SELECT * FROM alarm_table")
    LiveData<List<Alarm>> getAlarms();

    @Query("SELECT * FROM alarm_table WHERE id = :id")
    Alarm findItem(int id);

    @Update
    void update(Alarm alarm);
}

File: app/src/main/java/manabboro/alarmclock/database/AlarmDao.kt
package manabboro.alarmclock.database

import androidx.lifecycle.LiveData
import androidx.room.*
import manabboro.alarmclock.model.Alarm

@Dao
interface AlarmDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(alarm: Alarm?)

    @Query("DELETE FROM alarm_table")
    fun deleteAll()

    @get:Query("SELECT * FROM alarm_table")
    val alarms: LiveData<List<Alarm>>

    @Query("SELECT * FROM alarm_table WHERE id = :id")
    fun findItem(id: Int): Alarm

    @Update
    fun update(alarm: Alarm?)
}